realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -185,11 +188,25 @@ private void emitAccessors(JavaWriter writer) throws IOException {
                 writer.emitStatement(
                         "realm.checkIfValid()"
                 );
+                // Although setting null value for String and bytes[] can be handled by the JNI code, we still generate the same code here.
```

Comment:
Note that this is a departure from what we previously has done. While I agree with the fail-fast philosophy. We also have an upcoming dynamic API, where this error check is also relevant, so keeping this in Java might actually be counter productive.